gostream/ffmpeg/avformat.go

```diff
@@ -0,0 +1,343 @@
+//go:build cgo && !android
+
+package ffmpeg
+
+//#include <libavformat/avformat.h>
+//#include <libavutil/avutil.h>
+//#include <libavcodec/avcodec.h>
+// static AVCodecParameters * av_format_context_get_codec_par(AVFormatContext* fmt_ctx, int video_stream) {
+// 		return fmt_ctx->streams[video_stream]->codecpar;
+// }
+import "C"
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"unsafe"
+)
+
+type (
+	// Stream the AVStream
+	Stream C.struct_AVStream
+	// FormatContext AVFormatContext
+	FormatContext struct {
+		p  *C.struct_AVFormatContext
+		pp **C.struct_AVFormatContext
+	}
+	// Dictionary an AVDictionary
+	Dictionary struct {
+		p  *C.struct_AVDictionary
+		pp **C.struct_AVDictionary
+	}
+)
+
+// DictionaryEntries entries to be added to an ffmpeg.Dictionary
+type DictionaryEntries map[string]string
+
+// NewDictionary creates a new dictionary
+func NewDictionary(ctx context.Context, e DictionaryEntries) (Dictionary, error) {
+	var d *C.struct_AVDictionary
+	for k, v := range e {
+		select {
+		case <-ctx.Done():
+			return Dictionary{}, ctx.Err()
+		default:
+			if i, err := strconv.Atoi(v); err == nil {
+				if err := ErrorFromCode(int(C.av_dict_set_int(&d, C.CString(k), C.long(i), 0))); err != nil { //nolint: gocritic
```

Comment:
[note] There is an issue with `C.long` type on MacOS. It is expecting `C.longlong`. We may need to write some polyfill helper to sub c types based on platform.

Reply:
Oh yeah I see. It looks pretty simple to solve. One way would be to create a function in two separate files, guarded by build constraints that returns a long on non-Darwin, and a longlong on Darwin.